charmhub/client.go

```diff
@@ -19,170 +19,61 @@ import (
 	"context"
 	"fmt"
 	"io"
-	"net/http"
 	"net/url"
 	"path"
-	"sort"
 	"strings"
 	"time"
 
 	"github.com/juju/charm/v9"
 	"github.com/juju/errors"
-	"github.com/juju/loggo"
 
 	charmhubpath "github.com/juju/juju/charmhub/path"
 	"github.com/juju/juju/charmhub/transport"
 	charmmetrics "github.com/juju/juju/core/charm/metrics"
-	corelogger "github.com/juju/juju/core/logger"
-	"github.com/juju/juju/version"
 )
 
-// ServerURL holds the default location of the global charm hub.
-// An alternate location can be configured by changing the URL
-// field in the Params struct.
 const (
-	CharmHubServerURL     = "https://api.charmhub.io"
-	CharmHubServerVersion = "v2"
-	CharmHubServerEntity  = "charms"
-
-	MetadataHeader = "X-Juju-Metadata"
```

Comment:
It seems this is only in play when `WithMetadataHeaders` is used, and that looks to be unused right now. This is surprising as I thought the intent was to match what we do with charm store in sending anonymous stats like model uuid etc. So I think we need to retain this stuff and wire it up with metrics as we always have used.

Reply:
In this PR I think we should stick to simplifying rather than adding features. If we want to add stats/tracking we can do that in a subsequent PR. Does the new Charmhub API actually track those headers, or would they need to add the feature first? And then we'd need to add metrics dashboards, and so on ... seems out of scope for this refactoring.

Reply:
Metrics in charmhub is significantly different. We do not use X-Juju-Metdata at all there. I just didn't get a chance to pull it out when I made the change.

Reply:
Per video discussion, Heather mentioned we use `transport.RequestMetrics` for this now. See also [`charmhubRequestMetadata`](https://github.com/juju/juju/blob/fd886a18e169e4741f557d6b574bba06ff0b190d/apiserver/facades/controller/charmrevisionupdater/updater.go#L480).

Reply:
metrics are also gathered from request actions on request api calls when we use a unique persistent instance-key once a charm has actually be deployed and downloaded.